### PR TITLE
feat(AP-3446): replace http.request() with native fetch()

### PR DESCRIPTION
**Pull request by mattstrom**

### CODE REVIEW

File: src/helpers/requests/fetch-request.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      method: options.method,
+      headers: headers,
+    })
+      .then(function (res) {
+        if (res.status === 200) {
+          resolve(res.json());
+        } else if (res.status === 202) {
+          resolve();
+        } else {
+          const message = 'Evolv: Invalid status ' + res.statusCode + ' for response ' + res.statusMesssage;
```
Comment (DT-1236) [inline src/helpers/requests/fetch-request.js:37]: I see you changed `statusCode` to `status` for the other blocks. That should prolly apply here. IDE's also complaining about `statusMessage`. Is that the correct field?

File: src/helpers/requests/fetch-request.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import MiniPromise from '../../ponyfills/minipromise.js';
+import { createSignatureHeader } from './create-signature-header.js';
+
+
+const URL_PATTERN = /^([a-z]+):\/\/([^/]+)(.*)/i;
+
+
+export default function fetchRequest(options) {
+  return MiniPromise.createPromise(function(resolve, reject) {
```
Comment (DT-1236) [inline src/helpers/requests/fetch-request.js:9]: Is this library to like... add in promises to a js environment that doesn't have them?

File: src/helpers/requests/fetch-request.js (hunk 1 above)
Comment (mattstrom) [reply src/helpers/requests/fetch-request.js:37]: Hmm, that might have been Copilot's fault

File: src/helpers/requests/fetch-request.js (hunk 2 above)
Comment (mattstrom) [reply src/helpers/requests/fetch-request.js:9]: Yeah, that is the purpose. It's probably unnecessary now that we have ceased supporting IE11.

**Review by DT-1236 — APPROVED** [review]
A smidge unnerving that it's not typescript, but I suppose poking around in an IDE that's powered  by typescript will have to do

**Comment by mattstrom** [comment]
> A smidge unnerving that it's not typescript, but I suppose poking around in an IDE that's powered by typescript will have to do

Don't get me started about this 😞

**Pull request merged**